x-pack/plugins/snapshot_restore/__jest__/client_integration/home.test.ts

```diff
@@ -0,0 +1,707 @@
+              .textContent
+          ).toContain(`Remove repository '${rows[0].columns[1].value}'?`);
+        });
+
+        test('should send the correct HTTP request to delete repository', async () => {
+          const { component, table, actions } = testBed;
+          const { rows } = table.getMetaData('repositoryTable');
+
+          await actions.clickRepositoryActionAt(0, 'delete');
+
+          const modal = document.body.querySelector(
+            '[data-test-subj="deleteRepositoryConfirmation"]'
+          );
+          const confirmButton: HTMLButtonElement | null = modal!.querySelector(
+            '[data-test-subj="confirmModalConfirmButton"]'
+          );
+
+          await act(async () => {
+            confirmButton!.click();
+            await nextTick();
+            component.update();
+          });
+
+          const latestRequest = server.requests[server.requests.length - 1];
+          const repositoryName = rows[0].columns[1].value;
+
+          expect(latestRequest.method).toBe('DELETE');
+          expect(latestRequest.url).toBe(`/api/snapshot_restore/repositories/${repositoryName}`);
+        });
+      });
+
+      describe('detail panel', () => {
+        test('should show the detail when clicking on a repository', async () => {
+          const { exists, actions } = testBed;
+
+          expect(exists('repositoryDetail')).toBe(false);
+
+          await actions.clickRepositoryAt(0);
+
+          expect(exists('repositoryDetail')).toBe(true);
+        });
+
+        test('should set the correct title', async () => {
+          const { find, actions } = testBed;
+
+          await actions.clickRepositoryAt(0);
+
+          expect(find('repositoryDetail.title').text()).toEqual(repo1.name);
+        });
+
+        test('should show a loading while fetching the repository', async () => {
+          const { find, exists, actions } = testBed;
+
+          // By providing undefined, the "loading section" will be displayed
+          httpRequestsMockHelpers.setGetRepositoryResponse(undefined);
+
+          await actions.clickRepositoryAt(0);
+
+          expect(exists('repositoryDetail.sectionLoading')).toBe(true);
+          expect(find('repositoryDetail.sectionLoading').text()).toEqual('Loading repository…');
+        });
+
+        describe('when the repository has been fetched', () => {
+          beforeEach(async () => {
+            httpRequestsMockHelpers.setGetRepositoryResponse({
+              repository: {
+                name: 'my-repo',
+                type: 'fs',
+                settings: { location: '/tmp/es-backups' },
+              },
+              snapshots: { count: 0 },
+            });
+
+            await testBed.actions.clickRepositoryAt(0);
+          });
+
+          test('should have a link to the documentation', async () => {
+            const { exists } = testBed;
+
+            expect(exists('repositoryDetail.documentationLink')).toBe(true);
```

Comment:
@silne30 this is another good candidate for a functional test. Make sure we don't point to a 404 doc page if anything changes in `elastic.co` site.